### PR TITLE
Added force_download flag for downloading locally existing datasets

### PR DESCRIPTION
**Pull request by shreyansjainn**
# Description

Currently, if the dataset exists locally, the download_dataset method just raises a warning and downloads the dataset anyway as mentioned in #79 . This change introduces a flag `force_download` which by default will be `False` and will not download the dataset if a local copy exists. Users can specify the `force_download` flag as `True` to override the setting and download a fresh copy of the dataset. 

For integration with CLI, the flag is fixed at `True` as the CLI already has a similar check present. For unit tests, the flag is fixed at `True` for testing the `download_dataset` method.

Fixes # (issue), Depends on # (pull request)

## Type of change

> Please delete options that are not relevant.

- New feature (non-breaking change which adds functionality)

# Checklist:

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `pre-commit run --all-files` (see `CONTRIBUTING.md` instructions to set it up)
- [x] I have run `pytest -v` and no errors are present.
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I solved any possible warnings that `pytest -v` has generated that are related to my code to the best of my knowledge.
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: minari/storage/hosting.py [hunk 1]
```diff
@@ -76,14 +77,21 @@ def _upload_local_directory_to_gcs(local_path, bucket, gcs_path):
...
         dataset_id (str): name id of the Minari dataset
+        force_download (bool): boolean flag for force downloading the dataset. Default Value = False
     """
     file_path = get_dataset_path(dataset_id)
     if os.path.exists(file_path):
+        if not force_download:
+            logger.warn(
+                f"Dataset {dataset_id} found locally at {file_path}, Use force_download=True to download the dataset again.\n"
+            )
+            sys.exit()
```
Comment (younik) [inline minari/storage/hosting.py:93]: Why `sys.exit()` instead of an `if-else`?

File: minari/wrappers/error.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+"""Set of Error classes for gymnasium."""
+
+
+class Error(Exception):
```
Comment (RedTachyon) [inline minari/wrappers/error.py:4]: Do we need this? I know it exists in gymnasium which is already an interesting design choice (inherited from gym), but in that case it's probably better to just import it from there. Or just subclass `Exception` directly.

File: minari/storage/hosting.py [hunk 3]
```diff
@@ -76,14 +77,23 @@ def _upload_local_directory_to_gcs(local_path, bucket, gcs_path):
...
+def download_dataset(dataset_id: str, force_download: bool = False):
     """Download dataset from remote Farama server.
 
     Args:
         dataset_id (str): name id of the Minari dataset
+        force_download (bool): boolean flag for force downloading the dataset. Default Value = False
     """
     file_path = get_dataset_path(dataset_id)
     if os.path.exists(file_path):
+        if not force_download:
```
[the remark this answers is not in the document]
Comment (shreyansjainn) [reply minari/storage/hosting.py:89]: i kept the initial version as the one just raising the warning, but after recommendation from @balisujohn i changed it to raising an exception so it becomes more explicit. If the exception is not needed I can revert the changes back to just the warning also. 

https://github.com/Farama-Foundation/Minari/pull/90#discussion_r1228446166

[the remark this answers is not in the document]
Comment (balisujohn) [reply minari/storage/hosting.py:89]: I'll defer to @RedTachyon's opinion here, lets switch to a warning.

**Review by balisujohn — CHANGES_REQUESTED** [review]
Thanks for your work on this. I requested changes to the `download_dataset` function. Also please add a test asserting that the warning is correctly issued.

File: minari/storage/hosting.py [hunk 4]
```diff
@@ -76,14 +77,21 @@ def _upload_local_directory_to_gcs(local_path, bucket, gcs_path):
...
```
[the remark this answers is not in the document]
Comment (RedTachyon) [reply minari/storage/hosting.py:88]: Yes, that's precisely what I was thinking, sorry if that wasn't clear. If the dataset is downloaded, just end the function, inform the user what happened and carry on.

**Review by balisujohn — APPROVED** [review]

**Comment by balisujohn** [comment]
LGTM :^)

**Pull request merged**